packages/core-contracts/contracts/token/ERC721Enumerable.sol

```diff
@@ -0,0 +1,141 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ERC721.sol";
+import "./ERC721EnumerableStorage.sol";
+import "../interfaces/IERC721Enumerable.sol";
+
+/*
+    Reference implementations:
+    * OpenZeppelin - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721Enumerable.sol
+*/
+
+abstract contract ERC721Enumerable is ERC721, ERC721EnumerableStorage, IERC721Enumerable {
+    error IndexOutOfBounds();
+
+    function _initialize(
+        string memory tokenName,
+        string memory tokenSymbol,
+        string memory baseTokenURI
+    ) internal virtual override {
+        super._initialize(tokenName, tokenSymbol, baseTokenURI);
+        ERC721EnumerableStore storage store = _erc20EnumerableStore();
+        if (store.allTokens.length > 0) {
+            revert InitError.AlreadyInitialized();
+        }
```

Comment:
@leomassazza can you double check this?  

calling `super` here and calling the erc721 initialializer; and i'm ok if the store is reinitialized as long as there's no tokens created so thats the check on line 23.  

please suggest or shed light on anything im missing here.  thanks!

Reply:
initialization should only happens once. super._initialize will revert if tokenName, tokenSymbol or baseTokenURI was already set.

Your double checking makes sense here but not likely to revert

Reply:
you can simplify a little the code since store is not used anymore in that function and just get the allTokens directly:
if (_erc20EnumerableStore().allTokens.length >0) { ... }

Reply:
makes sense! thanks for feedback :)